azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourceGroupThenCreateNodes.java

```diff
@@ -62,35 +72,69 @@ protected CreateResourceGroupThenCreateNodes(
            GroupNamingConvention.Factory namingConvention,
            @Named(Constants.PROPERTY_USER_THREADS) ListeningExecutorService userExecutor,
            CustomizeNodeAndAddToGoodMapOrPutExceptionIntoBadMap.Factory customizeNodeAndAddToGoodMapOrPutExceptionIntoBadMapFactory,
-           AzureComputeApi api) {
+           AzureComputeApi api, AzureComputeServiceContextModule.AzureComputeConstants azureComputeConstants) {
       super(addNodeWithGroupStrategy, listNodesStrategy, namingConvention, userExecutor,
               customizeNodeAndAddToGoodMapOrPutExceptionIntoBadMapFactory);
       this.api = checkNotNull(api, "api cannot be null");
       checkNotNull(userExecutor, "userExecutor cannot be null");
+      this.azureComputeConstants = azureComputeConstants;
+      this.azureGroupName = this.azureComputeConstants.azureResourceGroup();
    }
 
    @Override
    public Map<?, ListenableFuture<Void>> execute(String group, int count, Template template,
                                                  Set<NodeMetadata> goodNodes, Map<NodeMetadata, Exception> badNodes,
                                                  Multimap<NodeMetadata, CustomizationResponse> customizationResponses) {
 
+      AzureTemplateOptions options = template.getOptions().as(AzureTemplateOptions.class);
+      // create resource group for jclouds group if it does not already exist
       ResourceGroupApi resourceGroupApi = api.getResourceGroupApi();
-      ResourceGroup resourceGroup = resourceGroupApi.get(group);
+      ResourceGroup resourceGroup = resourceGroupApi.get(azureGroupName);
       final String location = template.getLocation().getId();
-      final String resourceGroupName;
 
       if (resourceGroup == null){
-
          final Map<String, String> tags = ImmutableMap.of("description", "jClouds managed VMs");
-         resourceGroupName = resourceGroupApi.create(group, location, tags).name();
-      } else {
-         resourceGroupName = resourceGroup.name();
+         resourceGroupApi.create(azureGroupName, location, tags).name();
+      }
+
+      this.vnetName = this.azureGroupName + "virtualnetwork";
+      this.subnetName = this.azureGroupName + "subnet";
+
+      if (options.getVirtualNetworkName() != null) {
+         this.vnetName = options.getVirtualNetworkName();
```

Comment:
Don't use member variables (shared mutable state!) for things that depend on particular method execution parameters.